src/server/qtquick/private/wquickinputmethodv2_p.h

```diff
@@ -0,0 +1,235 @@
+#pragma once
+#include <wglobal.h>
+#include <qwglobal.h>
+#include <QObject>
+#include <QQmlEngine>
+#include <qwinputmethodv2.h>
+#include "wquickwaylandserver.h"
+#include "wtoplevelsurface.h"
+#include "wsurfaceitem.h"
+
+Q_MOC_INCLUDE("wquicktextinputv3_p.h")
+Q_MOC_INCLUDE("wsurface.h")
+
+struct wlr_input_method_v2_state;
+QW_BEGIN_NAMESPACE
+class QWInputMethodV2;
+class QWInputPopupSurfaceV2;
+class QWInputMethodKeyboardGrabV2;
+QW_END_NAMESPACE
+
+WAYLIB_SERVER_BEGIN_NAMESPACE
+class WQuickInputPopupSurfaceV2Private;
+class WQuickInputMethodKeyboardGrabV2Private;
+class WQuickInputMethodV2Private;
+class WInputDevice;
+class WSeat;
+class WQuickInputMethodV2;
+class WQuickInputMethodManagerV2Private;
+class WQuickTextInputV3;
+class WSurface;
+
+class WQuickInputMethodManagerV2 : public WQuickWaylandServerInterface, public WObject
+{
+    Q_OBJECT
+    QML_NAMED_ELEMENT(InputMethodManagerV2)
+    W_DECLARE_PRIVATE(WQuickInputMethodManagerV2)
+
+public:
+    explicit WQuickInputMethodManagerV2(QObject *parent = nullptr);
+
+Q_SIGNALS:
+    void beforeDestroy(WQuickInputMethodManagerV2 *self);
+    void inputMethod(WQuickInputMethodV2 *im);
+
+private:
+    void create() override;
+};
+
+class WQuickInputPopupSurfaceV2;
+class WInputPopupSurfacePrivate;
+class WInputPopupSurface : public WToplevelSurface, public WObject
+{
+    Q_OBJECT
+    W_DECLARE_PRIVATE(WInputPopupSurface)
+    QML_NAMED_ELEMENT(WaylandInputPopupSurface)
+    QML_UNCREATABLE("Only created in C++")
+
+public:
+    explicit WInputPopupSurface(WQuickInputPopupSurfaceV2 *surface, QObject *parent = nullptr);
+    bool doesNotAcceptFocus() const override;
+    WSurface *surface() const override;
+    bool isActivated() const override;
+    QRect getContentGeometry() const override;
+
+Q_SIGNALS:
+    void beforeDestroy(WInputPopupSurface *self);
+
+public Q_SLOTS:
+    bool checkNewSize(const QSize &size) override;
+};
+
+class WInputPopupSurfaceItem : public WSurfaceItem
+{
+    Q_OBJECT
+    Q_PROPERTY(WInputPopupSurface* surface READ surface WRITE setSurface NOTIFY surfaceChanged)
+    Q_PROPERTY(QPointF implicitPosition READ implicitPosition CONSTANT)
+    Q_PROPERTY(QSize minimumSize READ minimumSize CONSTANT FINAL)
+    Q_PROPERTY(QSize maximumSize READ maximumSize CONSTANT FINAL)
+    QML_NAMED_ELEMENT(InputPopupSurfaceItem)
+public:
+    explicit WInputPopupSurfaceItem(QQuickItem *parent = nullptr);
+
+    WInputPopupSurface *surface() const;
+    void setSurface(WInputPopupSurface *surface);
+
+    QPointF implicitPosition() const;
+    QSize minimumSize() const;
+    QSize maximumSize() const;
+
+Q_SIGNALS:
+    void surfaceChanged();
+
+private:
+    WInputPopupSurface *m_inputPopupSurface;
+
+    // WSurfaceItem interface
+protected slots:
```

Comment:
Why is slot?

Reply:
Generated by QtCreator. I've already corrected it.